### PR TITLE
Dashboard bug fixes and monitoring

### PR DESCRIPTION
**Pull request by SarahJaine**
## Description of change

- Adds https://github.com/adhocteam/Head-Start-TTADP/pull/376

> Altered the API and the frontend code so that each line on the Total Hours graph starts at the first date and proceeds through. If a report has 0 reports on that day, the graph reflects it. (There are no longer partial lines)
> 
> When an array of days without month prefixes is passed into the graph, plotly no longer auto converts them into numbers.
> 
> Language has been updated to say "Total Hours of TTA and Grants Served." The URL for for the Regional Analytics page was changed from /dashboard to /regional-analytics.
> 
> Internally to the application, the <Dashboard /> component has been renamed <RegionalAnalytics />

- Adds https://github.com/adhocteam/Head-Start-TTADP/pull/379

> We wanted to have an idea of what the user experience (response time and throughput) is with the widgets before we release the dashboard features to the main user group. Currently, the dashboard is only available to admin users.
> 
> Unfortunately New Relic isn't playing nicely with our node js set up, and the api routes are being grouped. Under APM > Transactions > you'll see the widget traces (and a lot of other api calls) appear under /* transaction.
> 
> This PR uses the new relic api to name the widget transactions so we can see their performance in new relic.

- Adds https://github.com/adhocteam/Head-Start-TTADP/pull/378

> I upgraded react-scripts which got rid of postcss and ws vulnerabilities.
> I safely ignored browserslist, glob-parent, and tar vulnerabilities

- Also now includes bug fixes found in demo, and camelCase instead of PascalCase from review https://github.com/adhocteam/Head-Start-TTADP/pull/380

## How to test
All changes demonstrated today or discussed. See individual feature branch PRs (links above in description section) for further testing info.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-258
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-261
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-268
* fixes for yarn audit failures

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/pages/RegionalDashboard/index.js [hunk 1]
```diff
@@ -16,16 +16,20 @@ import ReasonList from '../../widgets/ReasonList';
 import TotalHrsAndGrantee from '../../widgets/TotalHrsAndGranteeGraph';
 import './index.css';
 
-function Dashboard({ user }) {
-  const [appliedRegion, updateAppliedRegion] = useState(0);
-  const [regions, updateRegions] = useState([]);
-  const [regionsFetched, updateRegionsFetched] = useState(false);
+export default function RegionalDashboard({ user }) {
+  const hasCentralOffice = user && user.homeRegionId && user.homeRegionId === 14;
```
Comment (matthinz) [inline frontend/src/pages/RegionalDashboard/index.js:20]: It's early days for me and this codebase, but `14` stands out to me as a magic number here--a future refactor should consider introducing a helper function to tell whether a region has a central office or possibly creating a `regions.json` file for tracking these sorts of attributes.

File: frontend/src/widgets/__tests__/WidgetHelper.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import '@testing-library/jest-dom';
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import FormatNumber from '../WidgetHelper';
+
+const renderFormattedNumber = (number, decimalPlaces) => {
+  render(<div>{FormatNumber(number, decimalPlaces)}</div>);
+};
+
+describe('Format Number', () => {
```
Comment (matthinz) [inline frontend/src/widgets/__tests__/WidgetHelper.js:10]: 👍 to new tests!

**Review by matthinz — COMMENTED** [review]
👋 Hi! I'm new here, so my review mostly focused on a few code hygiene items and nitpicks. Overall looks good!

File: src/widgets/totalHrsAndGranteeGraph.js [hunk 3]
```diff
@@ -82,7 +92,7 @@ export default async function totalHrsAndGranteeGraph(scopes, query) {
       'startDate',
       'ttaType',
       'duration',
-      [sequelize.col('"activityRecipients->grant"."id"'), 'granteeId'],
+      // [sequelize.col('"activityRecipients->grant"."id"'), 'granteeId'],
```
[the remark this answers is not in the document]
Comment (SarahJaine) [reply src/widgets/totalHrsAndGranteeGraph.js:95]: That's usually my preference too. This is @AdamAdHocTeam 's code, but the reason why I didn't make the same comment in review was that we're bringing back this code in a ticket that's slated for _this_ sprint https://ocio-jira.acf.hhs.gov/browse/TTAHUB-216

[the remark this answers is not in the document]
Comment (matthinz) [reply src/widgets/totalHrsAndGranteeGraph.js:95]: Sorry, I don't have JIRA access yet--is that a sprint in the near future? I'm fine with leaving as-is if someone's going to be messing around with this code in the near future.

[the remark this answers is not in the document]
Comment (SarahJaine) [reply src/widgets/totalHrsAndGranteeGraph.js:95]: It's in the same sprint that we're in right now, 21.3.3. So yes, the _very_ near future.

[the remark this answers is not in the document]
Comment (matthinz) [reply src/widgets/totalHrsAndGranteeGraph.js:95]: Lol, works for me!

**Review by PatricePascual-ACF — APPROVED** [review]

File: src/widgets/totalHrsAndGranteeGraph.js [hunk 4]
```diff
@@ -91,6 +101,7 @@ export default async function totalHrsAndGranteeGraph(scopes, query) {
     },
     raw: true,
     includeIgnoreAttributes: false,
+    /*
```
[the remark this answers is not in the document]
Comment (SarahJaine) [reply src/widgets/totalHrsAndGranteeGraph.js:104]: resolved by https://github.com/HHS/Head-Start-TTADP/pull/519#discussion_r683773602

File: src/widgets/totalHrsAndGranteeGraph.test.js [hunk 5]
```diff
@@ -175,42 +175,28 @@ describe('Total Hrs and Grantee Graph widget', () => {
     const data = await totalHrsAndGranteeGraph(scopes, query);
 
     // Overall trace categories.
-    expect(data.length).toEqual(5);
+    expect(data.length).toEqual(4);
 
-    // Dates Placeholder.
+    // Grantee Rec TTA.
+    /*
```
[the remark this answers is not in the document]
Comment (SarahJaine) [reply src/widgets/totalHrsAndGranteeGraph.test.js:181]: resolved by https://github.com/HHS/Head-Start-TTADP/pull/519#discussion_r683773602

File: src/widgets/totalHrsAndGranteeGraph.test.js [hunk 6]
```diff
@@ -243,35 +229,24 @@ describe('Total Hrs and Grantee Graph widget', () => {
...
+    expect(data.length).toEqual(4);
 
     // Grantee Rec TTA.
-    expect(data[1].x.length).toEqual(3);
-    expect(data[1].y.length).toEqual(3);
-
-    expect(data[1].x).toEqual(['10', '15', '20']);
-    expect(data[1].y).toStrictEqual([2, 2, 4]);
-
+    /*
```
[the remark this answers is not in the document]
Comment (SarahJaine) [reply src/widgets/totalHrsAndGranteeGraph.test.js:235]: resolved by https://github.com/HHS/Head-Start-TTADP/pull/519#discussion_r683773602

File: src/widgets/totalHrsAndGranteeGraph.test.js [hunk 7]
```diff
@@ -303,35 +278,25 @@ describe('Total Hrs and Grantee Graph widget', () => {
...
-    expect(data.length).toEqual(5);
+    expect(data.length).toEqual(4);
 
     // Grantee Rec TTA.
-    expect(data[1].x.length).toEqual(3);
-    expect(data[1].y.length).toEqual(3);
-
-    expect(data[1].x).toEqual(['Jan', 'Feb', 'Mar']);
-    expect(data[1].y).toStrictEqual([2, 4, 2]);
+    /*
```
[the remark this answers is not in the document]
Comment (SarahJaine) [reply src/widgets/totalHrsAndGranteeGraph.test.js:284]: resolved by https://github.com/HHS/Head-Start-TTADP/pull/519#discussion_r683773602

File: src/widgets/totalHrsAndGranteeGraph.test.js [hunk 8]
```diff
@@ -368,41 +333,27 @@ describe('Total Hrs and Grantee Graph widget', () => {
     const data = await totalHrsAndGranteeGraph(scopes, query);
 
     // Overall trace categories.
-    expect(data.length).toEqual(5);
+    expect(data.length).toEqual(4);
 
-    // Dates Placeholder.
+    // Grantee Rec TTA.
+    /*
```
[the remark this answers is not in the document]
Comment (SarahJaine) [reply src/widgets/totalHrsAndGranteeGraph.test.js:339]: resolved by https://github.com/HHS/Head-Start-TTADP/pull/519#discussion_r683773602

File: frontend/src/widgets/TotalHrsAndGranteeGraph.js [hunk 9]
```diff
@@ -17,41 +17,21 @@ export function TotalHrsAndGranteeGraph({ data, dateTime }) {
...
-      data[0]: All Dates
-      data[1]: Grantee Rec TTA
+      data[0]: Grantee Rec TTA
       data[1]: Hours of Training
       data[2]: Hours of Technical Assistance
       data[3]: Hours of Both
     */
 
     const traces = [
+      /*
```
[the remark this answers is not in the document]
Comment (SarahJaine) [reply frontend/src/widgets/TotalHrsAndGranteeGraph.js:27]: resolved by https://github.com/HHS/Head-Start-TTADP/pull/519#discussion_r683773602

File: frontend/src/widgets/WidgetHelper.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
+/* WidgetHelper.js */
+/* Helper Functions for Widgets */
+function FormatNumber(numberToFormat, decimalPlaces = 0) {
```
[the remark this answers is not in the document]
Comment (SarahJaine) [reply frontend/src/widgets/WidgetHelper.js]: fixed- thanks!

**Review by PatricePascual-ACF — APPROVED** [review]

**Pull request merged**